tests/monitoring/monitoring.go

```diff
@@ -124,19 +126,24 @@ var _ = Describe("[Serial][sig-monitoring]Monitoring", Serial, decorators.SigMon
 			originalKv := util.GetCurrentKv(virtClient)
 			kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(context.Background(), originalKv.Name, metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
-			kv.Spec.CustomizeComponents = v1.CustomizeComponents{
-				Patches: []v1.CustomizeComponentsPatch{
-					{
-						ResourceName: virtHandler.deploymentName,
-						ResourceType: "DaemonSet",
-						Patch:        `{"spec":{"template":{"spec":{"nodeSelector":{"kubernetes.io/hostname":"does-not-exist"}}}}}`,
-						Type:         v1.StrategicMergePatchType,
+
+			patchData := map[string]interface{}{
+				"spec": map[string]interface{}{
+					"template": map[string]interface{}{
+						"spec": map[string]interface{}{
+							"nodeSelector": map[string]string{
+								"kubernetes.io/hostname": "does-not-exist",
+							},
+						},
 					},
 				},
 			}
 
+			patchBytes, err := json.Marshal(patchData)
+			Expect(err).ToNot(HaveOccurred())
+
 			Eventually(func() error {
```

Comment:
I would use json patch in this case. e.g.
```suggestion
			patchBytes, err := patch.New(WithAdd("/spec/template/nodeSelector/kubernetes.io~1hostname", "does-not-exist").GeneratePayload()
			Expect(err).ToNot(HaveOccurred())

			_, err = virtClient.KubeVirt(originalKv.Namespace).Patch(context.Background(), originalKv.Name, types.JSONPatchType, patchBytes, metav1.PatchOptions{})			
```

(note: `~1` is the go path replacement for "/")

Reply:
The whole idea is to get rid of the Eventually.